dali/kernels/imgproc/color_manipulation/brightness_contrast_gpu_test.cu

```diff
@@ -0,0 +1,149 @@
+// Copyright (c) 2019, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#include <gtest/gtest.h>
+#include <vector>
+#include <tuple>
+#include "dali/kernels/test/tensor_test_utils.h"
+#include "dali/kernels/test/kernel_test_utils.h"
+#include "dali/kernels/imgproc/color_manipulation/brightness_contrast_gpu.cuh"
+#include "dali/kernels/imgproc/color_manipulation/brightness_contrast_test_utils.h"
+
+namespace dali {
+namespace kernels {
+namespace brightness_contrast {
+namespace test {
+
+namespace {
+
+template <class In, class Out>
+std::enable_if_t<std::is_integral<Out>::value, Out> custom_round(float val) {
+  return std::round(val);//TODO round to nearest even
+}
+
+
+template <class In, class Out>
+std::enable_if_t<!std::is_integral<Out>::value, Out> custom_round(float val) {
+  return val;
+}
+
+
+}  // namespace
+
+template <class InputOutputTypes>
+class BrightnessContrastCudaKernelTest : public ::testing::Test {
```

Comment:
```suggestion
class BrightnessContrastCudaKernelTest;

template <typename Output, typename Input>
 class BrightnessContrastCudaKernelTest<std::pair<Output, Input>> : public ::testing::Test {
```
This gives you the following:
- the test cannot be instantiated with something else
- it assigns a meaning to the pair constituents, making InputOutputTypes superfluous
- you readily get In and Out type names inside the class - no need to extract them using dependent names, unless required in a derived class (e.g. the test body).